src/main/java/database/controller/DatabaseServiceRestController.java

```diff
@@ -0,0 +1,120 @@
+package database.controller;
+
+import database.entity.BaseEntity;
+import database.service.JsonDatabaseService;
+import io.javalin.config.JavalinConfig;
+import io.javalin.http.BadRequestResponse;
+import io.javalin.http.Context;
+import io.javalin.http.NotFoundResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.javalin.apibuilder.ApiBuilder.*;
+
+public class DatabaseServiceRestController {
+    private static final JsonDatabaseService databaseService = new JsonDatabaseService();
+
+    public void configureRouter(JavalinConfig config) {
+        config.router.apiBuilder(() ->
+                path("/api/v1/database", () -> {
+                    path("/{entityClass}", () -> {
+                        post(this::handleCreateTable);
+                        delete(this::handleDeleteTable);
+                        path("/{entities}", () -> {
+                            post(this::handleAddNewRecord);
+                            path("/filter", () -> get(this::handleGetByFilters));
+                            get(this::handleGetAllRecords);
+                            path("/{id}", () -> {
+                                put(this::handleUpdateRecord);
+                                delete(this::handleDeleteRecord);
+                                get(this::handleGetById);
+                            });
+                            delete(this::handleRemoveAllRecords);
+                        });
+                    });
+                }));
+    }
+
+    private void handleCreateTable(Context ctx) {
+        Class<? extends BaseEntity> entityClass = getClassFromPath(ctx);
+        ctx.json(databaseService.createTable(entityClass));
+    }
+
+    private void handleDeleteTable(Context ctx) {
+        Class<? extends BaseEntity> entityClass = getClassFromPath(ctx);
+        ctx.json(databaseService.deleteTable(entityClass));
+    }
+
+    private void handleAddNewRecord(Context ctx) {
+        Class<? extends BaseEntity> entityClass = getClassFromPath(ctx);
+        BaseEntity entity = ctx.bodyAsClass(entityClass);
+        ctx.json(databaseService.addNewRecordToTable(entity));
+    }
+
+    private void handleGetByFilters(Context ctx) {
+        Class<? extends BaseEntity> entityClass = getClassFromPath(ctx);
+        Map<String, Object> filters = new HashMap<>();
+
+        //TODO: change filter logic with generics
+        String fullName = ctx.queryParam("fullName");
```

Comment:
Please use `ctx.queryParamMap()` to get all query parameters. I'm going to change a bit the logic of `databaseService.getByFilters`, so for now let's just send this map to `databaseService.getByFilters` without any additional action. Please mark this method with **TODO: do not use this method since service method will be reworked**.